planet/api/models.py

```diff
@@ -36,52 +37,43 @@ class Request():
 
     :param url: URL of API endpoint
     :type url: str
-    :param auth: Planet API authentication
-    :type auth: :py:Class:'planet.auth.auth
-    :param params: values to send in the query string, defaults to None
+    :param params: Values to send in the query string. Defaults to None.
     :type params: dict, list of tuples, or bytes, optional
-    :param body_type: Expected response body type, defaults to `Body`
-    :type body_type: type, optional
-    :param data: object to send in the body, defaults to None
+    :param data: Object to send in the body. Defaults to None.
     :type data: dict, list of tuples, bytes, or file-like object, optional
-    :param method: HTTP request method, defaults to 'GET'
+    :param json: JSON to send. Defaults to None.
+    :type json: dict, optional
+    :param method: HTTP request method. Defaults to 'GET'
     :type method: str, optional
     :raises RequestException: When provided `body_type` is not a subclass of
         :py:class:`planet.api.models.Body`
     '''
-    def __init__(self, url, auth, params=None, body_type=None, data=None,
-                 method='GET'):
-        self.url = url
-        self.auth = auth
-        self.params = params
-
-        self.body_type = body_type or Body
-        if not issubclass(self.body_type, Body):
-            raise RequestException(
-                f'body_type ({self.body_type}) must be a subclass of Body'
-            )
+    def __init__(self, url, params=None, data=None, json=None, method='GET'):
+        if data or json:
+            headers = {'Content-Type': 'application/json'}
+        else:
+            headers = None
 
-        self.data = data
-        self.method = method
+        self.http_request = httpx.Request(
+            method,
+            url,
+            params=params,
+            data=data,
+            json=json,
+            headers=headers)
 
     @property
-    def headers(self):
-        '''Prepare headers for request.
+    def url(self):
+        return self.http_request.url
 
-        :returns: prepared headers
-        :rtype: dict
+    @url.setter
+    def url(self, url):
+        '''Set the url.
+
+        :param url: URL of API endpoint
+        :type url: str
         '''
-        headers = {}
-        if self.data:
-            headers['Content-Type'] = 'application/json'
-
-        if self.auth:
-            headers.update({
-                'Authorization': 'api-key %s' % self.auth.value
-            })
-        else:
-            raise exceptions.InvalidAPIKey('No API key provided')
-        return headers
+        self.http_request.url = httpx.URL(url)
 
 
 class Response():
```

Comment:
This class seems like it could be gotten rid of now that `_body` is gone. :)

Reply:
I prefer to have an interface between our codebase and the one we rely on. aka, I like to have a `Response` class that wraps the `httpx.Response` class. This is good practice in general, although it is particularly important since `httpx` is still beta, very advanced beta but beta, so it is not guaranteed that the API will not change.

Reply:
In that case, what about having it be a subclass of `httpx.Response`?

Reply:
I do not see how subclassing will provide for a clean interface that is easy to adapt to changes in `httpx.Response`

Reply:
Okay, so then what about using that interface in our tests, so instead of using `httpx.Response` we can use this class?

Reply:
Agree with @jreiberkyle on this, wrapping the may-change-in-the-future Response is the best for future users; subclassing doesn't address future adaptability. This is fine as-is.